src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -160,6 +158,17 @@ pub fn cstr_to_str(b: &Bytes) -> Result<&str, Utf8Error> {
     std::str::from_utf8(without_null)
 }
 
+/// Record `sql` in the current tracing span.
+fn record_sql_in_span(sql: &str) {
```

Comment:
What's the relationship between this and the `run_stmt` in #10660?

Reply:
I assume you're talking about the span created here:

https://github.com/risingwavelabs/risingwave/blob/9dcad747a462f9dd1df571f8812ffa5f36596197/src/utils/pgwire/src/pg_protocol.rs#L230-L236

This replaces the old one, which was renamed to `process_query_msg_one_stmt` after that PR. Pros:

- It's defined in the unified path for handling `Query` `Parse` and `Execute`, while the old one only handled (simple) `Query`.
- It's now not only used by traces, but also used to provide common info for the query logs later, including the slow ones.
https://github.com/risingwavelabs/risingwave/blob/9dcad747a462f9dd1df571f8812ffa5f36596197/src/utils/pgwire/src/pg_protocol.rs#L312-L316
https://github.com/risingwavelabs/risingwave/blob/9dcad747a462f9dd1df571f8812ffa5f36596197/src/utils/pgwire/src/pg_protocol.rs#L288-L292

As for recording the SQL field after creating the span: I just don't want to duplicate the code for extracting the SQL from the messages. This also reduces allocations.